### ISSUE TITLE
ELF symbol versions: `SymbolVersionDefinition` with multiple entries do not get correctly linked to their `SymbolVersion`

### ISSUE THREAD

**Issue by aeflores** [lief-project/LIEF#749]
A `SymbolVersionDefinition` can contain multiple `SymbolVersionAux` entries. However, a `SymbolVersion` object can only hold one `SymbolVersionAux`. As a result, the code that assigns the `SymbolVersionAux` from a `SymbolVersionDefinition` to the corresponding `SymbolVersion` repeatedly writes on the same field https://github.com/lief-project/LIEF/blob/master/src/ELF/Parser.tcc#L1530 and only the last entry in the `SymbolVersionDefinition` is associated with the symbol version.

**To Reproduce**

You can reproduce the bug with the following example:

foo.c
```
#include <stdio.h>

__asm__(".symver foo2,foo@LIBFOO_2.0");
void foo2(void)
{
    puts("foo 2.0");
}
```
Compiled with linker script `libfoo.map`
```
LIBFOO_1.0 {
    global:
      foo;
    local:
      *;
};

LIBFOO_2.0 {
    global:
      foo;
    local:
      *;
} LIBFOO_1.0;
```
The compilation command would be:
```
gcc foo.c -o libfoo.so -shared  -fPIC -Wl,--version-script=libfoo.map
```
Then we can load the library with lief (python API)
```
>>> foo_bin= lief.parse("libfoo.so")
>>> foo_sym = [sym for sym in foo_bin.dynamic_symbols if sym.name == "foo"][0]
>>> foo_sym.symbol_version.symbol_version_auxiliary.name
'LIBFOO_1.0'
```

What is happening is that the symbol `foo` is associated with a `SymbolVersionDefinition` that has two entries `LIBFOO_2.0` and `LIBFOO_1.0`. Since the second entry is `LIBFOO_1.0`, that is the one that get associated to the symbol version in the end.

Note that other tools will report the first entry instead of the last one. This is the case of `readelf` e.g.:
```
>> readelf libfoo.so -aW | grep foo
6: 000000000000060a    19 FUNC    GLOBAL DEFAULT   13 foo@LIBFOO_2.0
```

**Expected behavior**

I think the solution is for `SymbolVersion` to be able to point to a `SymbolVersionDefinition` that might have multiple entries or to contain a list of `SymbolVersionAux` directly. I suspect a similar issue can happen with `SymbolVersionRequirement` since those can have multiple entries as well. This of course would be a breaking change since it would change the public API.

**Environment (please complete the following information):**
 - System and Version : Ubuntu 18.04
 - Target format ELF
 - LIEF commit version: 0.12.1-c7b3ce3b

**Additional context**

**Comment by aeflores** [comment]
I've continued looking into this. It looks like `SymbolVersionDefinition` can contain multiple `SymbolVersionAux` entries, but in all the cases I've been able to find, the first entry is the symbol version itself and the second entry is the parent version. This can be seen with `readelf libfoo.so --version-info` which yields:
```
Version definition section '.gnu.version_d' contains 3 entries:
  Addr: 0x00000000000003a0  Offset: 0x0003a0  Link: 4 (.dynstr)
  000000: Rev: 1  Flags: BASE  Index: 1  Cnt: 1  Name: libfoo2.so
  0x001c: Rev: 1  Flags: none  Index: 2  Cnt: 1  Name: LIBFOO_1.0
  0x0038: Rev: 1  Flags: none  Index: 3  Cnt: 2  Name: LIBFOO_2.0
  0x0054: Parent 1: LIBFOO_1.0
```
Here the symbol version definition `LIBFOO_2.0` has two entries. The first one is `LIBFOO_2.0` itself, and the second one is the parent ` LIBFOO_1.0` (as defined in the linker script). In general, it looks like all entries except the first one are just predecessor versions (see e.g. https://akkadia.org/drepper/symbol-versioning explanation of the vda_next field):

>The first entry (pointed
    to by the Elfxx_Verdef entry, contains the actual defined name.  The
    second and all later entries name predecessor versions.

Therefore, for many uses, I believe that reporting the first entry of the symbol definition in the `SymbolVersion` should be sufficient. That would NOT require an API change, which is good news. If anyone needs more detailed information, they can access the list of `SymbolVersionDefinition` directly.
